pkg/ivd/ivd_protected_entity_type_manager_util.go

```diff
@@ -8,41 +8,164 @@ import (
 	"github.com/vmware/govmomi/cns"
 	cnstypes "github.com/vmware/govmomi/cns/types"
 	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/property"
 	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/mo"
 	vim25types "github.com/vmware/govmomi/vim25/types"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"strings"
 )
 
-func findAllDatastores(ctx context.Context, client *vim25.Client) ([]vim25types.ManagedObjectReference, error) {
+func findHostsOfNodeVMs(ctx context.Context, client *vim25.Client, config *rest.Config, logger logrus.FieldLogger) ([]vim25types.ManagedObjectReference, error) {
+	// #1: get hostNames of all node VMs
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeList, err := clientSet.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	vmHostNameMap := make(map[string]bool)
+	for _, node := range nodeList.Items {
+		if node.Name == "" {
+			return nil, errors.New("One of the node VM in the cluster has empty node name")
```

Comment:
Can we output the UID here?  It would be nice to associate the node that is having a problem with something.  Actually, can we use UID instead of name for all of this?

Reply:
We cannot use UID as a key here. Since k8s node UID doesn't have any correlation with VM in VC. Added UID in the error message.